src/ui/public/vislib/lib/axis.js

```diff
@@ -21,9 +23,13 @@ export default function AxisFactory(Private) {
       this.el = args.el;
       this.xValues = args.xValues;
       this.ordered = args.ordered;
-      this.xAxisFormatter = args.xAxisFormatter;
+      this.axisFormatter = args.type === 'category' ? args.xAxisFormatter : args.yAxisFormatter;
       this.expandLastBucket = args.expandLastBucket == null ? true : args.expandLastBucket;
       this._attr = _.defaults(args._attr || {});
+      this.scale = null;
+      this.domain = [args.yMin, args.yMax];
+      this.elSelector = args.type === 'category' ? '.x-axis-div' : '.y-axis-div';
+      this.type = args.type;
```

Comment:
Maybe we can move this up, and use `this.type` rather than `args.type` throughout